### ISSUE TITLE
Keybindings: Can't use "enter" to cancel "actions.find" (search.action.cancel?)

### ISSUE THREAD

**Issue by amelio-vazquez-reina** [microsoft/vscode#87261]
Issue Type: <b>Bug</b>

I have the following in my `keybindings.json`
```
    {
        "key": "ctrl+g",
        "command": "search.action.cancel",
        "when": "listFocus && searchViewletVisible && !inputFocus"
    },
    {
        "key": "enter",
        "command": "search.action.cancel",
        "when": "listFocus && searchViewletVisible && !inputFocus"
    },    
    {
        "key": "escape",
        "command": "-search.action.cancel",
        "when": "listFocus && searchViewletVisible && !inputFocus"
    },
```

but when I invoke `actions.find`, type a string to search and then press `enter`, it doesn't do anything (i.e. it keeps the search widget open). However, `ctrl+g` does work (it cancels the search).

VS Code version: Code - Insiders 1.42.0-insider (26f5dfcd76c9a02f116a706844611aeb09ae8c0b, 2019-12-18T10:32:01.697Z)
OS version: Darwin x64 19.2.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz (16 x 2300)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|64.00GB (34.60GB free)|
|Process Argv|-psn_0_2310708|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (32)</summary>

Extension|Author (truncated)|Version
---|---|---
atlascode|atl|2.2.1
vscode-database|baj|2.2.0
path-intellisense|chr|1.4.2
bracket-pair-colorizer|Coe|1.0.61
vscode-markdownlint|Dav|0.33.0
gitlens|eam|10.2.0
LogFileHighlighter|emi|2.6.0
shell-format|fox|6.1.3
seito-openfile|Fr4|1.8.5
mdmath|goe|2.4.0
gc-excelviewer|Gra|2.1.32
open-file-from-path|jac|1.3.3
latex-workshop|Jam|8.5.0
vsc-space-block-jumper|jmf|1.2.2
center-editor-window|kai|2.3.0
vscode-docker|ms-|0.9.0
python|ms-|2019.11.50794
sqltools|mtx|0.21.5
neuron-IPE|neu|1.0.4
vscode-ipython|pan|0.5.0
advanced-new-file|pat|1.2.0
vscode-versionlens|pfl|0.24.0
vscode-yaml|red|0.6.1
vscode-dired|rru|0.0.4
code-settings-sync|Sha|3.4.3
vscode-fileutils|sle|3.0.0
emacs-mcx|tut|0.13.0
vscodeintellicode|Vis|1.2.0
vscode-icons|vsc|9.6.0
org-mode|vsc|1.0.0
highlight-trailing-white-spaces|yba|0.0.2
markdown-all-in-one|yzh|2.6.1

</details>

**Comment by alexdima** [comment]
@amelio-vazquez-reina To clarify, are you focused in the search across files input?

**Comment by amelio-vazquez-reina** [comment]
@alexdima I'm referring to search within a file, that is, **not** the viewlet to the left. Thanks.

**Comment by alexdima** [comment]
@amelio-vazquez-reina `search.action.cancel` is an action that cancels the search across files. The search inside a file cannot be canceled. The search widget can be closed via the command `closeFindWidget`.

None of the keybindings you have written match when the editor find widget is focused because the context keys `listFocus && searchViewletVisible` do not hold when the find widget is focused. Those context keys hold when the search viewlet is visible and the focus is in a list in the UI.

**Comment by amelio-vazquez-reina** [comment]
Thanks @alexaubry maybe I'm misunderstanding your comment. 

You said:

> The search inside a file cannot be canceled

but and **I can** actually cancel it (as I said in my OP) using `ctrl+g`. What I'm unable to do is bind this to `enter` (e.g. I found what I want, enter = proceed to the next step, _exiting_ search). 

Looking at my `keybindings.json` I see that I have `ctrl+g` also bound to `editor.cancelOperation` as well. Could that be the command that is getting triggered to cancel search within files?

**Comment by alexdima** [comment]
I have looked at the implementation of `editor.cancelOperation` from [here](https://github.com/microsoft/vscode/blob/c7cc30ca5aa4f64c1f7d85343594a7801029e8d4/src/vs/editor/browser/core/keybindingCancellation.ts#L89-L106) and have run multiple find references to figure out who is registering cancellation tokens to that service. 

The result is: format, go to definition, go to declaration, go to type definition, go to implementation, find all references, rename and the outline. 

So all of the actions I listed above end up registering their requests with the `IEditorCancellationTokens` service, which can then cancel the requests if `editor.cancelOperation` is called. 

This does not include the find widget. The find results in the find widget are computed synchronously (without a request to rich language services) [here](https://github.com/microsoft/vscode/blob/c7cc30ca5aa4f64c1f7d85343594a7801029e8d4/src/vs/editor/common/model/textModel.ts#L1070)

So I don't believe that `editor.cancelOperation` is clearing the find widget. Perhaps you have even more things bound to `ctrl+g`? 

Also, to clarify, here is the find widget:
![image](https://user-images.githubusercontent.com/5047891/73051095-2f646280-3e82-11ea-9efb-01227372ec9b.png)
